encoding/flag/flag.go

```diff
@@ -0,0 +1,170 @@
+package flag
+
+import (
+	"encoding"
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/iancoleman/strcase"
+)
+
+const (
+	flagTag = "flag"
+	descTag = "comment" // do we want a different tag for the flag vs toml?
+	fmtTag  = "fmt"
+)
```

Comment:
does the "config" tag need to be supported?

Reply:
yes - but it can be a separate pr.